### ISSUE TITLE
Build errors inside unicore?

### ISSUE THREAD

**Issue by iandoesallthethings** [biappi/muScribble#2]
I'm trying to flash this firmware onto a cheapo stm32f4 (weact brand? Not sure if it matters) but I'm getting stuck on the build step. I've been stumbling through getting all the dependencies squared away, but it seems like unicore-mx has some weird syntax errors. I've tried to debug it, but I'm pretty unfamiliar.

```bash
➜  muScribble git:(master) ✗ make flash
make -C unicore-mx TARGETS=stm32/f4 FP_FLAGS="-mfloat-abi=soft"
  GENHDR  include/unicore-mx/sam/3x/irq.json
  GENHDR  include/unicore-mx/sam/4l/irq.json
  GENHDR  include/unicore-mx/sam/3s/irq.json
  GENHDR  include/unicore-mx/sam/3a/irq.json
  GENHDR  include/unicore-mx/sam/3u/irq.json
  GENHDR  include/unicore-mx/sam/3n/irq.json
  GENHDR  include/unicore-mx/lpc17xx/irq.json
  GENHDR  include/unicore-mx/lpc13xx/irq.json
  GENHDR  include/unicore-mx/lpc43xx/m0/irq.json
  GENHDR  include/unicore-mx/lpc43xx/m4/irq.json
  GENHDR  include/unicore-mx/stm32/f4/irq.json
  GENHDR  include/unicore-mx/stm32/f3/irq.json
  GENHDR  include/unicore-mx/stm32/l4/irq.json
  GENHDR  include/unicore-mx/stm32/f2/irq.json
  GENHDR  include/unicore-mx/stm32/l0/irq.json
  GENHDR  include/unicore-mx/stm32/f0/irq.json
  GENHDR  include/unicore-mx/stm32/f7/irq.json
  GENHDR  include/unicore-mx/stm32/l1/irq.json
  GENHDR  include/unicore-mx/stm32/f1/irq.json
  GENHDR  include/unicore-mx/nrf/51/irq.json
  GENHDR  include/unicore-mx/nrf/52/irq.json
  GENHDR  include/unicore-mx/lm3s/irq.json
  GENHDR  include/unicore-mx/efm32/gg/irq.json
  GENHDR  include/unicore-mx/efm32/g/irq.json
  GENHDR  include/unicore-mx/efm32/lg/irq.json
  GENHDR  include/unicore-mx/efm32/tg/irq.json
  GENHDR  include/unicore-mx/vf6xx/irq.json
  BUILD   lib/stm32/f4
/bin/sh: -c: line 0: unexpected EOF while looking for matching `"'
/bin/sh: -c: line 1: syntax error: unexpected end of file
make[1]: *** [Makefile:70: lib/stm32/f4] Error 2
make: *** [Makefile:47: unicore-mx/lib/libucmx_stm32f4.a] Error 2
```

Any idea what's going wrong here?

Edit: For the record, I also tried building in ubuntu running inside parallels. Similar errors, but slightly different messages. I'm not sure if the fact that it's apple silicon/arm would matter since they're syntax errors. I don't currently have an intel machine to try it on.

Edit: just had the thought that it could be a line ending thing maybe? I'll try windows if I can get parallels working.

**Comment by iandoesallthethings** [comment]
Update: Trying it on debian 11 and getting the same errors, so I don't think it's a macos problem.

**Comment by biappi** [comment]
hi! I'm having issues as well on my personal machine (an arm mac as well), albeit different ones... I think unicore-mx has a bit bit rotten over the years unfortunately :( I tried to quickly swap in libopencm3, but the usb stack is different enough that's not a direct recompilation. 

I currently don't have much time to try and port this, but have you maybe looked into hacking the application code to work with an HAL library that works on your platform?  as long as you have midi input and spi output you should be able to hack something working

**Comment by iandoesallthethings** [comment]
> but have you maybe looked into hacking the application code to work with an HAL library that works on your platform? as long as you have midi input and spi output you should be able to hack something working

Yeah, I'm interested in giving it a go! I'm pretty unfamiliar with this space (I'm a web dev by trade), so I'm unsure what to look for in a HAL. But if it's just a matter of replacing anything that uses unicore-mx with something else that does the same thing, I feel like I can probably figure it out.

My only other shortcut is to ask if you happen to have an old compiled binary laying around and hope my chip (stm32f411ce) is similar enough

Edit: lol this is hard 
![noidea](https://github.com/biappi/muScribble/assets/15148313/af4e2673-5348-47a9-b4e9-7ac303d8357d)
